### PR TITLE
Integration tests

### LINKED ISSUE #87: Integration tests
We need integration test "framework" in virtlet and tests for the features we have already.

Items:
- [x] "framework"
- [x] image service - by using fake grcp request
- [x] virtualization service - by using fake grpc request
- [x] k8s integration - running local k8s cluster with virtlet

### PR DESCRIPTION
**Pull request by ivan4th**
Moving @nhlfr's work from #88 here.

Fixes #87 

Items:
- [x] "framework"
- [x] image service - by using fake grcp request
- [x] virtualization service - by using fake grpc request
- [x] k8s integration - running local k8s cluster with virtlet

---
This change is [Reviewable](https://reviewable.io/reviews/mirantis/virtlet/113)

### CODE REVIEW

File: contrib/images/local-cluster/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+# Fake docker executable is only needed to fool local-up-cluster, we
+# don't really need docker inside the container
+RUN apt-get update && \
+    DEBIAN_FRONTEND=noninteractive apt-get install -y sudo rsync && \
+    apt-get clean && \
+    rm -rf /var/lib/apt/lists/* /tmp/* && \
+    mkdir -p /go/src/k8s.io && \
+    echo -e '#!/bin/sh\nexit 0' >/usr/bin/docker && chmod +x /usr/bin/docker && \
+    go get -u github.com/jteeuwen/go-bindata/go-bindata && \
+    git clone -b v1.5.0-alpha.1 https://github.com/kubernetes/kubernetes.git /go/src/k8s.io/kubernetes
```
Comment (jellonek) [inline contrib/images/local-cluster/Dockerfile]: Maybe this version could be passed as env variable?

**Comment by ivan4th** [comment]
Review status: 0 of 27 files reviewed at latest revision, 1 unresolved discussion.

---

*[contrib/images/local-cluster/Dockerfile, line 14 at r2](https://reviewable.io:443/reviews/mirantis/virtlet/113#-KWitHearFqNq_kUdtBN-r2-14:-KWr6voURlaLApVdiGDv:b303bpe) ([raw file](https://github.com/mirantis/virtlet/blob/5c85a3cd46e712e9bc0aa03f83de6217cc0170a1/contrib/images/local-cluster/Dockerfile#L14)):*
<details><summary><i>Previously, jellonek (Piotr Skamruk) wrote…</i></summary>
> Maybe this version could be passed as env variable?
</details>
Updated it to use build args

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/mirantis/virtlet/113)*

**Comment by jellonek** [comment]
:lgtm_strong: 

---

Reviewed 7 of 13 files at r1, 21 of 26 files at r2, 2 of 2 files at r3, 2 of 3 files at r4.
Review status: 26 of 27 files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/mirantis/virtlet/113#-:-KWrBXyYvP3OE4xhwRtr:b-uwkowd)*

**Comment by pigmej** [comment]
Reviewed 7 of 13 files at r1, 21 of 26 files at r2, 2 of 2 files at r3, 3 of 3 files at r4.
Review status: all files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/mirantis/virtlet/113)*

**Pull request merged**